torch_geometric/loader/neighbor_loader.py

```diff
@@ -512,3 +538,25 @@ def to_index(tensor):
             num_nodes = feature_store.get_tensor_size(input_nodes)[0]
             return node_type, range(num_nodes)
         return node_type, input_nodes.index
+
+
+def get_mixed_sampling_nodes(data: HeteroData,
+                             input_nodes: List[InputNodes]) -> SamplingNodes:
+    for i in range(len(input_nodes)):
+        input_nodes[i] = get_sampling_nodes(data, input_nodes[i])
+    return to_hetero_types(input_nodes), to_hetero_list(input_nodes)
+
+
+def to_hetero_types(input_nodes: List[Tuple[str, Tensor]]) -> List[str]:
+    return [node_type for node_type, index in input_nodes]
+
+
+def to_hetero_list(input_nodes: List[Tuple[str, Tensor]]) -> HeteroNodeList:
+    return [(node_type, i) for node_type, index in input_nodes for i in index]
```

Comment:
There might be cases where we return nodes of only one type. Should we mention that this can't be used to train a model which predicts on multiple node types?

Reply:
Hmm I didn't fully understand this - do you mean some samples from the neighbor sampler will only contain a single type?

Reply:
Yeah it might contain samples from only one type. 
I am not sure what the use case of this multiple node sampling is. If an envisioned use case is is predicting on multiple node types in a hetero graph, that might not work.